src/index.spec.ts

```diff
@@ -1,12 +1,227 @@
+    expect(isInteger(-0.5)).toBeFalsy();
+  });
+});
+
+describe('toLowerCase test', () => {
+  it('should return abc with ABC', () => {
+    expect(toLowerCase('ABC')).toMatch(/abc/);
+  });
+  it('should return 23hF with 23HF', () => {
+    expect(toLowerCase('23HF')).toMatch(/23hf/);
+  });
+  it('should return 23hF with 23hf', () => {
+    expect(toLowerCase('23hf')).toMatch(/23hf/);
+  });
+  it('should return abc with abc', () => {
+    expect(toLowerCase('abc')).toMatch(/abc/);
+  });
+  it('should return Please provide a string with empty string', () => {
+    expect(toLowerCase('')).toBe('Please provide a string');
+  });
+});
+
+describe('removeDuplicatesFromArray test', () => {
+  it('should return [1] with [1]', () => {
+    expect(removeDuplicatesFromArray([1])).toEqual([1]);
+  });
+  it('should return [1,"a",3] with [1,"a",3]', () => {
+    expect(removeDuplicatesFromArray([1, 'a', 3])).toEqual([1, 'a', 3]);
+  });
+  it('should return [1,2,3] with [1,2,3,2]', () => {
+    expect(removeDuplicatesFromArray([1, 2, 3, 2])).toEqual([1, 2, 3]);
+  });
+  it('should return Error with any type diferent to string[] or number[]', () => {
+    const testFunction = () => {
+      removeDuplicatesFromArray({} as any);
+    };
+    expect(testFunction).toThrow(Error);
+  });
+});
+
+describe('createProduct test', () => {
+  it('should return a product with a objetc verified with joi', () => {
+    const product1 = {
+      name: 'Browneggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const resp = {
+      id: expect.any(Number),
+      name: 'Browneggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    expect(createProduct(product1)).toEqual(resp);
+  });
+  it('should return Error when object doesnt verified with joi', () => {
+    const product2 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const product3 = {
+      name: 'Brown eggs',
+      tags: {} as any,
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const product4 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs this is extra text for this field',
+      price: 28.1,
+    };
+    const product5 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 1,
+    };
+    const testFunction2 = () => {
+      createProduct(product2);
+    };
+    const testFunction3 = () => {
+      createProduct(product3);
+    };
+    const testFunction4 = () => {
+      createProduct(product4);
+    };
+    const testFunction5 = () => {
+      createProduct(product5);
+    };
+    expect(testFunction2).toThrow(Error);
+    expect(testFunction3).toThrow(Error);
+    expect(testFunction4).toThrow(Error);
+    expect(testFunction5).toThrow(Error);
```

Comment:
This should be on different `it()` statements, with descriptive test names and error messages.